mlos_bench/mlos_bench/environments/composite_env.py

```diff
@@ -82,6 +86,18 @@ def __init__(self,
         if not self._children:
             raise ValueError("At least one child environment must be present")
 
+    def __enter__(self) -> Environment:
+        self._contexts = [env.__enter__() for env in self._children]
+        return super().__enter__()
+
+    def __exit__(self, ex_type: Optional[Type[BaseException]],
+                 ex_val: Optional[BaseException],
+                 ex_tb: Optional[TracebackType]) -> Literal[False]:
+        for env in reversed(self._children):
+            env.__exit__(ex_type, ex_val, ex_tb)
```

Comment:
What happens if one of these raises an exception, but not all of them?
Is there retry logic somewhere that (maybe in the future) could get implemented?
Would this all or nothing cleanup of the list of existing contexts cause problems?

Reply:
ooooh, that's right! Really nice catch! Will fix it now